shared/aggregation/maxcover.go

```diff
@@ -0,0 +1,130 @@
+package aggregation
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/go-bitfield"
+)
+
+// ErrInvalidMaxCoverProblem is returned when Maximum Coverage problem was initialized incorrectly.
+var ErrInvalidMaxCoverProblem = errors.New("invalid max_cover problem")
+
+// MaxCoverProblem defines Maximum Coverage problem.
```

Comment:
Can you add some documentation for godocs that explains the Maximum Coverage problem?